source/sasl-auth.rst

```diff
@@ -0,0 +1,246 @@
+.. _sasl:
+
+================================================================================
+Setting up LDAP authentication using SASL 
+================================================================================
+
+This document describes an example configuration
+suitable only to test out the external authentication functionality
+in a non-production environment.
+Use common sense to adapt these guidelines to your production environment.
+
+To learn more about how the authentication works, see :ref:`ldap-authentication-sasl`. 
+
+Environment setup and configuration
+===================================
+
+The following components are required:
+
+* ``slapd``: OpenLDAP server.
+* ``libsasl2`` version 2.1.25 or later.
+* ``saslauthd``: |SASL| Authentication Daemon (distinct from ``libsasl2``).
+
+The following steps will help you configure your environment:
+
+.. contents::
+   :local:	  
+
+.. rubric:: Assumptions
+
+Before we move on to the configuration steps, we assume the following:
+
+1. You have the LDAP server up and running. The LDAP server is accessible to the server with |PSMDB| installed. This document focuses on OpenLDAP server. If you use Microsoft Windows Active Directory, refer to the :ref:`windows-ad` section for ``saslauthd`` configuration. 
+#. You must place these two servers behind a firewall as the communications between them will be in plain text. This is because the SASL mechanism of PLAIN can only be used when authenticating and credentials will be sent in plain text.
+#. You have ``sudo`` privilege to the server with the |PSMDB| installed.
+
+Configuring ``saslauthd``
+---------------------------------
+1. Install the SASL packages. Depending on your OS, use the following command:
+
+   For *RedHat and CentOS*:
+
+   .. code-block:: bash
+
+      $ sudo yum install -y cyrus-sasl cyrus-sasl-plain
```

Comment:
For 4.0.26, 4.2.16, 4.4.8, 5.0.2 (PSMDB-873) cyrus-sasl-plain package isn't required, it's installed as a dependency